lang/cs/Org.Apache.REEF.Client/Common/HttpClientHelper.cs

```diff
@@ -90,12 +108,17 @@ internal static List<string> GetRmUri(string filePath)
             }
         }
 
-        internal static string GetAppId(string filePath)
+        /// <summary>
+        /// Get application id for the Job submitted 
+        /// </summary>
+        /// <param name="filePath"></param>
+        /// <returns></returns>
+        internal string GetAppId(string filePath)
         {
             using (var sr = new StreamReader(File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.Read)))
             {
-                var appId = sr.ReadLine();                
-                return appId;
+                _appId = sr.ReadLine();
```

Comment:
The change here makes the return value an instance variable. That is not the original behavior. Now. reads to `_appId` will return different things depending on when they are called.